### DISCUSSION TITLE
How to learn kha in 2021?

### DISCUSSION THREAD

**Discussion by VanBandama** [Kode/Kha#1355] [Q&A]
Hi! I'm new to Kha and eager to learn it. But all tutorials seems to be outdated. I mean when I run F1 and type init Kha project then structure of that project differs from what I saw on Lewis Lepton video tutorials. In his tutorials he got 2 files. main.hx and project.hx but I don't have it. What do I miss? Or this is just new way how kha handles it's files?

**Comment by RobDangerous** [comment]
Lewis' tutorials are years old, of course some things changed. But you can still use his tutorials - just don't get caught up in silly details like what files are generated by default and read through https://github.com/Kode/Kha/wiki first for up to date basics.

**Reply by VanBandama** [reply]
Well I tried and I have failed. Now Kode studio won't run kha from git. When I hit f5 on fresh init project everything passes but no window. Oddly enough when I comment out in user settings path to kha from git and run kha from kode's default path then window will pop up. 
Also when I copy paste no matter what example code (either Lewis's or khashmup or yet another) I got lot's of errors no matter what version of kha I use. What could be the cause?

Edit.
I removed Kode studio and downloaded fresh binaries from github releases page. But when I launch that fresh unpacked one it behaves like old one. Even paths to Kha in extensions remained. I was doing everything step by step carefully from this video - https://www.youtube.com/watch?v=U3HZF44DGwI&list=PL4neAtv21WOmmR5mKb7TQvEQHpMh1h0po&index=95

One major difference is I'm on windows

**Reply by RobDangerous** [reply]
Use vscode with the extension pack instead of Kode Studio - Kode Studio needs an update to work with the latest Kha, I didn't get to that yet.
And about the errors you got - can't say anything about that without you posting the actual errors.
Also again, please don't use Lewis' videos like that. Use them to learn what you can do, what features exist - but not to copy&paste code or follow things step by step. Also keep in mind that Kha is not for beginners. Happy when beginners use it but that's not the focus.

**Reply by VanBandama** [reply]
thanks for answering so quick. I'll do that, definitely ;)

**Reply by ohnarov** [reply]
@VanBandama I started with Kha just to create something super simple and fast. And I feel similar to you. Kha looks simple at first glance but it's not when you try to setup IDE for example. 
You need to know how to work with CLR, git and some IDE. 
No fresh step by step tutorials are exist at all ^)

So I decided to start with VS Code IDE + Haxe Extension Pack. Test projects Run with no problem. You can try any examples from Github and YouTube but be ready to fix outdated code.

**Reply by melMass** [reply]
I feel dumb but I don't understand how to build a Kha project with VScode's extension pack.
Also is it possible to build a Kha project from CLI alone?
I've used Kha (with KodeStudio) in the past but I can't remember anything

**Reply by VanBandama** [reply]
both on vscode (assuming you installed all in one kha pack) and kode studio with F1 (init kha project). Then next to green play button you will got two options. First is build to html5, and second is krom builder. Html5 project is located inside build folder in your kha project. For krom I didn't test it yet

**Reply by RobDangerous** [reply]
That's building and debugging. If you just want to build the project, hit ctrl+shift+b to get a list of all build task - that's standard vscode behavior. To build from the command line run node path/to/Kha/make --compile and if you want to build for a specific system run something like node path/to/Kha/make ios --compile.

**Reply by melMass** [reply]
Thanks!

For some reason, my task templates does not have Kha or Haxe specific, I will reinstall all to make sure
![image](https://user-images.githubusercontent.com/7041726/130271118-2f891aca-f4e5-4814-a1c6-bb158ee6a3ff.png)

The debugging config have specific ones for Haxe but nothing for Kha
![image](https://user-images.githubusercontent.com/7041726/130271261-672a2779-d08d-4783-a4cb-cb7fd3a7bc90.png)

**Reply by RobDangerous** [reply]
Create an empty directory, open it in vscode, hit F1, select "Kha: Init Project" and from then on things will work automatically, also when you later re-open that directory. There is no need to fill in task templates or create debug environments manually, once the extension detects that a Kha project is opened, everything appears automatically.
Or just follow http://kha.tech/getstarted and ignore that it's for Kode Studio instead of vscode.

**Reply by melMass** [reply]
Perfect Thanks!!

![image](https://user-images.githubusercontent.com/7041726/130284691-54cf5b20-b315-4a2d-bfa2-bee6853c275c.png)
